src/simple_server/core.clj

```diff
@@ -14,21 +14,68 @@
 ;;; and our "http code".  Our game now lives in its own namespace, and
 ;;; is fully testable independent of our "presentation layer".
 
-(defn new-game-handler []
-  (when (game/new-game!)
-    (response "OK- start guessing at /guess")))
-
-(defn guess-handler [guess]
-  (condp = (game/guess-answer guess)
-    nil       (-> (response  "You need to supply a guess with /guess?guess=N")
-                  (status 400))
-    :game-over (response  "Congratulations! You win!")
-    :too-low   (response "Too low.")
-    :too-high  (response  "Too high.")))
+;; An abstraction for accounts in the database
+(def users (atom #{"ray"}))
+
+;; An abstraction for currently logged in accounts
+(def current-users (atom #{}))
+
+(defn landing-handler []
+  ;; The landing page of the game, with instructions to tell user to either sign up or log in
+  (response (str "Welcome to Guessing the Number! "
+                 "Head to /sign-up?account=\"your account\" to sign up if this is your first time visiting. "
+                 "Or go to /log-in?account=\"your account\" to log in to start playing the game!")))
+
+(defn sign-up-handler [account]
+  ;; Check if the user already exists, if not, create it, otherwise, tell the user to log in
+  (if (nil? (some #{account} @users))
+    (do
+      (swap! users conj account)
```

Comment:
Awesome idea!  You're the only one who thought of maintaining a DB of existing users... nice!